SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -306,6 +306,15 @@ void USpatialNetDriver::OnConnectionToSpatialOSSucceeded()
 	Connection = ConnectionManager->GetWorkerConnection();
 	check(Connection);
 
+	// If the current Connection comes from an outdated ClientTravel, the associated NetDriver (this) won't match
+	// the NetDriver from the Engine, resulting in a crash. Here, if the NetDriver is outdated, we leave the callback.
+	if (bConnectAsClient && GEngine->GetWorldContextFromPendingNetGameNetDriver(this) == nullptr)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Outdated NetDriver connection skipped. May be due to an outdated ClientTravel"));
+		ConnectionManager->OnConnectedCallback.Unbind();
```

Comment:
let's also unbind the failure case (OnConnectionToSpatialOSFailed). It probably doesn't matter too much, but I think it would be nice to keep either both bound or both unbound